ssg/build_yaml.py

```diff
@@ -1331,13 +1333,42 @@ def __init__(
         self.stig_references = None
         if stig_reference_path:
             self.stig_references = ssg.build_stig.map_versions_to_rule_ids(stig_reference_path)
+        self.rule_to_components = self._load_components()
+
+    def _load_components(self):
+        if "components_root" not in self.env_yaml:
+            return None
+        product_dir = self.env_yaml["product_dir"]
+        components_root = self.env_yaml["components_root"]
+        components_dir = os.path.join(product_dir, components_root)
+        components = ssg.components.load(components_dir)
+        rule_to_components = ssg.components.get_rule_to_components_mapping(
+            components)
+        return rule_to_components
 
     def _process_values(self):
         for value_yaml in self.value_files:
             value = Value.from_yaml(value_yaml, self.env_yaml)
             self.all_values[value.id_] = value
             self.loaded_group.add_value(value)
 
+    def __process_rule(self, rule):
+        if self.rule_to_components is not None and rule.id_ not in self.rule_to_components:
```

Comment:
This is motivated mainly by feedback to contributors that forget to assign components, and this case is not caught by tests. A build failure provides an early feedback, and the contributor will have to think of components early on.

However, there are disadvantages as well:

- We may end up with an "other" component, but its rules would not share reasons to change, which is the main driving force behind components.
- Although there is an existing division by package related to files that the check/remediation operates on, this structure again may not be beneficial.
- If a rule should be assigned to more components, this error alone won't assure that.